modules/core/shared/src/main/scala/weaver/Test.scala

```diff
@@ -40,6 +40,6 @@ object Test {
 
   def apply[F[_]](name: String, f: F[Expectations])(
       implicit F: EffectCompat[F]
-  ): F[TestOutcome] = apply(name, (_: Log[F]) => f)
+  ): F[TestOutcome] = apply[F](name, (_: Log[F]) => f)
```

Comment:
We get the following error in 2.12, resolved by adding these type parameters:
```
[warn] compiler bug: Unexpected code path: testing two type variables for subtype relation:
[warn]   ?F <:< ?F
[warn] Please report bug at https://github.com/scala/bug/issues
````